x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlProcessorAutoscalingDecider.java

```diff
@@ -106,18 +107,30 @@ private boolean hasUnsatisfiedDeployments(TrainedModelAssignmentMetadata trained
         return trainedModelAssignmentMetadata.modelAssignments()
             .values()
             .stream()
+            .filter(deployment -> deployment.getTaskParams().getPriority() == Priority.NORMAL)
             .anyMatch(deployment -> deployment.isSatisfied(mlNodeIds) == false);
     }
 
     private MlProcessorAutoscalingCapacity.Builder computeRequiredCapacity(TrainedModelAssignmentMetadata trainedModelAssignmentMetadata) {
         int maxThreadsPerAllocation = 0;
         int processorCount = 0;
+        boolean hasLowPriorityDeployments = false;
         for (TrainedModelAssignment assignment : trainedModelAssignmentMetadata.modelAssignments().values()) {
+            if (assignment.getTaskParams().getPriority() == Priority.LOW) {
+                hasLowPriorityDeployments = true;
+                continue;
+            }
             int threadsPerAllocation = assignment.getTaskParams().getThreadsPerAllocation();
             maxThreadsPerAllocation = Math.max(maxThreadsPerAllocation, threadsPerAllocation);
             processorCount += assignment.getTaskParams().getNumberOfAllocations() * threadsPerAllocation;
         }
 
+        if (hasLowPriorityDeployments) {
+            // If there are low priority deployments let us ensure there will at least be one node required.
+            maxThreadsPerAllocation = Math.max(1, maxThreadsPerAllocation);
+            processorCount = Math.max(1, processorCount);
```

Comment:
This might cause us to scale beyond the smallest possible ESS node when the only thing running is one low priority model, for example if the smallest possible ESS node has 0.75 processors.

I think it's only necessary to change one of these to guarantee a node.

A possible solution would be to make `processorCount` a `double` and use `Math.max(0.1, processorCount)` here (not bothering to adjust `maxThreadsPerAllocation`.

Reply:
Well spotted, done.